### DISCUSSION TITLE
Logs are reporting “HTTP request returned status code 403:” When importing with the CLI.

### DISCUSSION THREAD

**Discussion by guillaumesoucy94** [Kovah/LinkAce#418] [Support]
Hello, 

I’m not able to import the half of the links from an .html file generated by Mozilla Firefox, the logs are reporting a lot of “HTTP request returned status code 403:”

This is an example: 

“https://www.rapidtables.com/calc/electric/electricity-calculator.html: https://www.rapidtables.com/calc/electric/electricity-calculator.html is not reachable. HTTP request returned status code 403:”

When visiting the site by entering the link in Mozilla Firefox, I’m able to visit the website without any issues.

How to bypass this HTTP 403 error? Using tag –skip-meta-generation in my syntax doesn’t change anything.

About the environment, I’m using the latest LinkAce version running on Ubuntu Server 20.04 LTS. Everything is up-to-date on the machine.

Thank-you, 

Guillaume

**Comment by Kovah** [comment]
Hi there.
There is probably not much you can do about that. LinkAce does not use a regular browser to access the websites to generate titles and descriptions, therefore some website block all "non-human" visits. That's why you see a 403 in LinkAce but access the site in your browser without issues.
You can try to add regular User Agents to your configuration, see [this documentation](https://github.com/Kovah/laravel-html-meta#configuration). But it's probably more about your server IP address, some sites block all visits from hosting providers.

**Reply by guillaumesoucy94** [reply]
Hello,

Yes, I'm getting the same issues sometimes when downloading websites, I remember that I had to add a user agent to my syntax to be able to downloads those websites.

Will try for sure to setup User Agent in LinkAce. 

For the server who host LinkAce, I'm hosting it locally in my own server room. It have it's own dedicated internet link, I think it should be fine as it use "Regular" business internet. 

(Had issues in the pass when getting content from Youtube with a server located at an external hosting provider. Their IP ranges was probably "rate limited" by Google...)

I'm trying the User Agent "fix" now and will post here what's happened right after.

Thanks again!

Guillaume

**Comment by guillaumesoucy94** [comment]
Hello, 

After adding the User Agents I still have issues with the import of a 3370 items .html file. 

The output throw no errors:

`Reading file "bookmarks_2022-03-20_22-43-19.html"...
Skipping link check.
0 links imported successfully, 3370 skipped.`

I'm not able to find all my links on the web interface.  

This is my config/html-meta.php file:

`<?php
return [
    'timeout' => 10,
    'parser' => \Kovah\HtmlMeta\HtmlMetaParser::class,
    'user_agents' => [
    'Mozilla/5.0 (Windows NT 6.4) AppleWebKit/537.36.0 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36.0',
    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12) AppleWebKit/…4.1.28 (KHTML, like Gecko) Version/15.2.0 Safari/604.1.28',
    'Mozilla/5.0 (compatible; Googlebot/2.1.0; +http://www.google.com/bot.html)',
    ],
];`

Is there something wrong with it?

Thank-you!

Guillaume

**Reply by Kovah** [reply]
I can't tell what's wrong with the links you imported. Are you sure that you did not delete any links or so? Because when importing bookmarks, LinkAce only skips those URLs that are already present in the database.
